### ISSUE TITLE
[Slider] LabelFor does not work

### ISSUE THREAD

**Issue by DavidCorrado** [material-components/material-components-android#2347]
**Description:** 
Per the description for accessibility for Slider it says to use label for:
https://material.io/components/sliders/android#using-sliders
If you do this it does not read the label for as the other views do.

**Expected behavior:**
Talkback should announce like it does with other views such as seekbar.  
It currently announces as:
0% Slider use volume keys to adjust
Should announce as
0% Slider for [TextView Text] use volume keys to adjust
So something like 

**Source code:** 
When having XML like this
`         <com.google.android.material.slider.RangeSlider
                    android:id="@+id/slider"
                    android:layout_width="match_parent"
                    android:layout_height="500dp"
                    android:stepSize="1.0"
                    android:valueTo="5.0" />
                <TextView
                    android:id="@+id/tv_slider"
                    android:layout_width="match_parent"
                    android:layout_height="wrap_content"
                    android:labelFor="@+id/slider"
                    android:text="Zero" />`

**Android API version:** Android API version here
30 Android 11
**Material Library version:** Material Android Library version you are using here (e.g., 1.1.0-alpha07)
1.4.0
**Device:** Device on which the bug was encountered here
Pixel 5

To help us triage faster, please check to make sure you are using the [latest version](https://github.com/material-components/material-components-android/releases) of the library.

We also happily accept [pull requests](https://github.com/material-components/material-components-android/pulls).

**Comment by FannyDemey** [comment]
Hi ! Any update about this issue ? 
Thanks
